src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/changes/PrincipalAddressChange.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.changes;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import uk.gov.companieshouse.api.model.company.RegisteredOfficeAddressApi;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.AddressDto;
+
+public class PrincipalAddressChange extends Change {
+    private static final String CHANGE_NAME = "changeOfRoa";
+
+    @JsonProperty("registeredOfficeAddress")
+    private RegisteredOfficeAddressApi registeredOfficeAddress;
```

Comment:
both registeredOfficeAddress and proposedRegisteredOfficeAddress need to have the same Type. That type needs to have the fields in the ODP https://companieshouse.atlassian.net/wiki/spaces/ROE2/pages/4144005155/ODP-ROE-Update-012-B+Change+list+Calculation#Changes-to-the-OE-entity%3A. Need to map from OverseasEntityDto - for proposed value and from Public/PrivateAPI for original/current value.